src/Post/Filter/TypeFilter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Post\Filter;
+
+use Flarum\Filter\FilterInterface;
+use Flarum\Filter\FilterState;
+
+class TypeFilter implements FilterInterface
+{
+    public function getFilterKey(): string
+    {
+        return 'type';
+    }
+
+    public function filter(FilterState $FilterState, string $filterValue, bool $negate)
```

Comment:
nitpick: can you please change all occurrences of `$FilterState` to `$filterState`

Reply:
... I thought I clicked that button in my find+replace tool... oh well...